drivers/gpu/drm/vc4/vc4_dsi.c

```diff
@@ -1538,7 +1538,9 @@ static int vc4_dsi_bind(struct device *dev, struct device *master, void *data)
 	 * so set up a channel for talking to it.
 	 */
 	if (dsi->port == 1) {
-		dsi->reg_dma_mem = dma_alloc_coherent(dev, 4,
+		dev_warn(dev,"Count %d: dsi->r.d.mem=0x%08x, dsu->r.d.chan=%d\n",callcount,(unsigned int)dsi->reg_dma_mem, (unsigned int)dsi->reg_dma_chan);
+		if (!dsi->reg_dma_mem)			//Consider this conditionial to prevent memory leak		
```

Comment:
I think this is the wrong approach:
1. An unsuccessful bind should probably have no effect on the system state. No additional action should be required to clean up.
2. An unbind should reverse the effects of the bind.

1) requires that error paths result in the releasing of any resources previously claimed during the bind attempt. This may end up duplicating code that appears in unbind, or it may be possible to factor it out into a "clean-up" function that is called from both places.

2) is not satisfied here because nowhere in the code is a call to e.g. `dma_release_channel`.

Reply:
Thank you for the feedback and helpful setting out of the principles to which one one should adhere - they make absolute sense.
I will re-do the changes in the way you suggest - probably early next week.